src/batSplay.ml

```diff
@@ -769,6 +802,40 @@ struct
     | Empty -> raise Not_found
     | Node (l, kv, r) -> kv, sref (bst_append l r)
 
+
+  let add_seq s m =
+    BatSeq.fold_left
+      (fun m (k, v) -> add k v m)
+      m
+      s
+    
+  let of_seq s =
+    add_seq s empty
+
+  let rec seq_of_iter m () =
+    match m with
+    | End -> BatSeq.Nil
+    | More(k, v, r, e) ->
+       BatSeq.Cons ((k, v), seq_of_iter (cons_enum r e))
+      
+  let to_seq m =
+    seq_of_iter (cons_enum (sget m) End)
+
+  let to_seq_from k m =
+    seq_of_iter (cons_enum_from k (sget m) End)
+   
+  let union f m1 m2 =
+    fold
+     (fun k v m ->
+        match find_opt k m with
+        | Some v1 ->
+           (match f k v v1 with
+            | Some vmerged -> add k vmerged m
+            | None -> remove k m)
+        | None -> add k v m)
+      m1
+      m2
```

Comment:
This implementation is not completely satisfying (it does not preserve the knowledge that we build the result in-order), but here the advice to use `merge` instead is not so convincing, because `merge` is okay but not that nice. (It performs an imperative traversal of two `Enum`, instead of working on the nicer, underlying `'a enumeration` structures.) I propose to leave it as-is in the PR, and I maybe submit a subsequent PR to improve `merge` and then define `union` from it.